src/encoded/static/components/filegallery.js

```diff
@@ -2916,6 +3002,10 @@ class FileGalleryRendererComponent extends React.Component {
             compiledAnalyses: compileAnalyses(props.analyses, datasetFiles, 'choose analysis'),
             /** Index of currently/last selected `compiledAnalyses`. */
             selectedAnalysesIndex: 0,
+            /** True if object is of type Series */
+            isSeriesType,
+            /** Series Files */
+            seriesFiles,
```

Comment:
I don’t think `isSeriesType` and `seriesFiles` shouldn’t exist in state because state shouldn’t serve as a cache. I do realize why you do this, but I hope we have a simpler solution. One is I think I made a mistake in `FileGallery` and shouldn’t have passed `data` in as its own state, but should pass it in directly, that way `data` in this constructor gets the series files passed in instead of extracting them from analyses (you can still use context.analyses for further filtering).

Reply:
I removed these.